net/libfab/libfab.c

```diff
@@ -1279,8 +1283,17 @@ static void libfab_buf_done(struct m0_fab__buf *buf, int rc)
 	 */
 	if (!fab_buf_tlink_is_in(buf)) {
 		/* Try to finalise. */
-		if (m0_thread_self() == &ma->ftm_poller)
+		if (m0_thread_self() == &ma->ftm_poller) {
+			if (buf->fb_length == (sizeof(uint64_t) * 2)) {
+				tmp = (uint64_t *)nb->nb_buffer.ov_buf[0];
+				if (*tmp == FAB_DUMMY_DATA) {
+					tmp++;
+					pas_buf = (struct m0_fab__buf *)(*tmp);
+					libfab_buf_complete(pas_buf, 0);
```

Comment:
Is this UT-specific code?

Reply:
No, this is not UT specific code, 
This is added to release buffer if remote completion is received in buffer other than dummy buffer.